src/renderer/api-decorator.js

```diff
@@ -357,7 +354,16 @@ limitations under the License.
         //---------------------------------------------------------------
         let winOpts = getWindowOptionsSync();
 
-        showOnReady(glbl, winOpts);
+        // Prevent iframes from attempting to do windowing actions, these will always be handled
+        // by the main window frame.
+        if (!window.frameElement) {
+            showOnReady(glbl, winOpts);
+        }
+
+        // The api-ready event allows the webContents to assign api priority. This must happen after
+        // any spin up windowing action or you risk stealing api priority from an already connected frame
+        electron.remote.getCurrentWebContents(renderFrameId).emit('openfin-api-ready', renderFrameId);
```

Comment:
are we sure this can be executed before `showOnReady` is "finished" for non-iframe window?

Reply:
for iframes the show on ready does not get called, so yeah

Reply:
I meant for non-iframe windows, "ready" will be emitted _before_ the "onFinish" callbacks in `showOnReady`.... is that an intentional change? doesn't sound right that we would emit "ready" before min/max/show calls are completed

Reply:
All the min/max/show functions in `showOnReady` are sync. `showOnReady` can actually be written in a more linear way now that everything is sync (this wasn't the case a while ago).